testbeam_analysis/tools/kalman.py

```diff
@@ -152,12 +159,63 @@ def _filter(transition_matrices, observation_matrices, transition_covariances,
     kalman_gains = np.zeros((chunk_size, n_timesteps, n_dim_state, n_dim_obs))
     filtered_states = np.zeros((chunk_size, n_timesteps, n_dim_state))
     filtered_state_covariances = np.zeros((chunk_size, n_timesteps, n_dim_state, n_dim_state))
+    # array where new transition matrices are stored, needed to pass it to kalman smoother
+    transition_matrices_update = np.zeros_like(transition_covariances)
 
     for t in range(n_timesteps):
         if t == 0:
```

Comment:
Just a generell comment. You access the first and second value with list[n-1], list[n] for `n >= 1`. It is slightly better coding if you would do list[n], list[n+1] for `n>=0`. Because `n-1 < 0` still works due to Python slicing rules, but  `n >= j` for a list with `j` elemtents would throw an `IndexError`. You want to have the `IndexError`, otherwise bugs can be hidden.